### PR TITLE
Implement admin api cassandra mechanism

### PR DESCRIPTION
**Pull request by hjmodi**

### CODE REVIEW

**Review by ligangty — COMMENTED** [review]
Do we really need the full repository model class like ArtifactStore, RemoteRepository, HostedRepository and Group? I think the tracking service can just create a cut version of these classes like how metadata-service and promote-service did to retrieve the needed info which will be used by tracking.

File: src/main/java/org/commonjava/indy/service/tracking/data/cassandra/RemoteKojiStoreDataCache.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+@Qualifier
+@Target( { ElementType.FIELD, ElementType.PARAMETER, ElementType.METHOD } )
+@Retention( RetentionPolicy.RUNTIME )
+@Documented
+public @interface RemoteKojiStoreDataCache
+{
+}
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/tracking/data/cassandra/RemoteKojiStoreDataCache.java:31]: Is this still needed?

File: src/main/java/org/commonjava/indy/service/tracking/data/cassandra/CassandraFoloUtil.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+    // the attributes of group repository
+    public static final String CONSTITUENTS = "constituents";
+
+    public static final String PREPEND_CONSTITUENT = "prependConstituent";
+
+    public static final String KEY = "key";
+
+    public static final String AFFECTED_STORES = "affectedStores";
+
+    public static final int MODULO_VALUE = 10;
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/tracking/data/cassandra/CassandraFoloUtil.java:89]: Seems lots of these constants are not serving for folo. Can we remove them?

File: src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        {
+            newParams[i] = String.valueOf( object );
+            i++;
+        }
+
+        this.params = newParams;
+        return this;
+    }
+
+}
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java:99]: I'm thinking if these IndyLifecycleException and WorkflowException are still needed in folo. Maybe a new FoloException is only needed for folo's purpose

File: src/main/java/org/commonjava/indy/service/tracking/model/AccessChannel.java [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+    /** Used when the store is accessed via httprox addon. */
+    GENERIC_PROXY,
+
+    /** Used to signify content coming from normal repositories and groups. */
+    NATIVE,
+
+    /** Used when the store is accessed via regular Maven repo.
+     *  NOTE: This has been changed to {@link #NATIVE} in our tracking code. It is included for historical purposes. */
+    @Deprecated MAVEN_REPO
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/tracking/model/AccessChannel.java:34]: Maybe this should be removed in new folo service.

File: src/main/java/org/commonjava/indy/service/tracking/data/cassandra/RemoteKojiStoreDataCache.java (hunk 1 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/data/cassandra/RemoteKojiStoreDataCache.java]: no, removed

File: src/main/java/org/commonjava/indy/service/tracking/data/cassandra/CassandraFoloUtil.java (hunk 2 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/data/cassandra/CassandraFoloUtil.java]: removed unnecessary variables.

File: src/main/java/org/commonjava/indy/service/tracking/model/AccessChannel.java (hunk 4 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/model/AccessChannel.java]: removed.

File: src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java (hunk 3 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java:99]: maybe we can revisit this after all the new code is merged in?

Comment (ligangty) [reply src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java:99]: Ok. So maybe a "TODO" is needed here to tell that this is needed to remove or replace.

File: src/main/java/org/commonjava/indy/service/tracking/model/pkg/PackageTypes.java [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    public static boolean contains( final String packageType )
+    {
+        return PACKAGE_TYPES.containsKey( packageType );
+    }
+
+    public static PackageTypeDescriptor getPackageTypeDescriptor( final String packageType )
+    {
+        return PACKAGE_TYPES.get( packageType );
+    }
+}
```
Comment (ligangty) [inline src/main/java/org/commonjava/indy/service/tracking/model/pkg/PackageTypes.java:75]: Maybe these "PackageTypes" and "PackageTypeDescriptor" are also not needed in folo? I think what is needed are in PackageTypeConstants.

File: src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java (hunk 3 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/exception/IndyLifecycleException.java:99]: added todo

File: src/main/java/org/commonjava/indy/service/tracking/model/pkg/PackageTypes.java (hunk 5 above)
Comment (hjmodi) [reply src/main/java/org/commonjava/indy/service/tracking/model/pkg/PackageTypes.java]: removed it.

**Review by ligangty — APPROVED** [review]
LGTM

**Pull request merged**